src/components/forms/CreateCase.tsx

```diff
@@ -3,103 +3,150 @@ import { useAppDispatch } from '../../store/hooks'
 import { createCase, openCaseForm } from '../../store/user/caseSlice'
 import { toast } from 'react-toastify'
 import TextField from '../textField'
+import { store } from '../../store/store'
+import instance from '../../api/axios.api'
+import { Form, useNavigate } from 'react-router-dom'
 
-const CreateCase: FC = () => {
-  const dispatch = useAppDispatch()
 
-  const [nameCase, setNameCase] = useState("")
-  const [descriptionCase, setDescriptionCase] = useState("")
-  const [drillDepthCase, setDrillDepthCase] = useState<number>(0)
-  const [pipeSizeCase, setPipeSizeCase] = useState<number>(0)
-  const [createdAtCase, setCreatedAtCase] = useState<Date>(new Date)
+interface ICaseForm {
+  type: "put" | "post";
+  id?: string;
+  prevName: string;
+  prevDescription: string;
+  prevDrillDepth: number;
+  prevPipeSize: number;
+  trajectoryId: string;
+  setIsEdit?: (edit: boolean) => void;
+  onSuccess?: () => void
+}
+
+const CreateCase: FC<ICaseForm> = ({type, id, prevName, prevDescription, prevDrillDepth, prevPipeSize, trajectoryId, setIsEdit, onSuccess}) => {
+  const navigate = useNavigate()
+
+
+  const [nameCase, setNameCase] = useState(prevName)
+  const [descriptionCase, setDescriptionCase] = useState(prevDescription)
+  const [drillDepthCase, setDrillDepthCase] = useState<number>(prevDrillDepth)
+  const [pipeSizeCase, setPipeSizeCase] = useState<number>(prevPipeSize)
   
 
-  const caseCreateFormOpenHandler = () => {
-    dispatch(openCaseForm())
+  const getTrajectoryIdFromStore = (): string | null => {
+    const state = store.getState();
+    return state.case.trajectoryId; 
   }
 
-  const createCaseHandle = () => {
-    console.log("It works 1")
-    dispatch(createCase({
-      id: "",
-      caseName: nameCase,
-      caseDescription: descriptionCase,
-      drillDepth: drillDepthCase,
-      pipeSize: pipeSizeCase,
-      createdAt: createdAtCase
-    }))
-    toast.success('Case was successfully created')
-    caseCreateFormOpenHandler()
+  const handleSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+    if (type == 'post') {
+      const newCase = {
+        case_name: nameCase,
+        case_description: descriptionCase,
+        drill_depth: drillDepthCase,
+        pipe_size: pipeSizeCase
+      };
+        
+      try {
+        const trajectoryId = getTrajectoryIdFromStore()
+        toast.success("ID: " + trajectoryId)
+        await instance.post(`/api/v1/cases?trajectoryId=${trajectoryId}`, newCase);
+        toast.success("Новый кейс был добавлен");
```

Comment:
instead of using try catch, then you would do something like this:
`TrajectoryService.create().then(()=>toast.success("Новый кейс был добавлен")).catch(()=>toast.success("Ошибка при добавлении кейса"))`